js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/AssetSidebarNode.tsx

```diff
@@ -43,164 +29,181 @@ export const AssetSidebarNode = ({
   explorerPath: ExplorerPath;
   onChangeExplorerPath: (path: ExplorerPath, mode: 'replace' | 'push') => void;
   onFilterToGroup: (group: AssetGroup) => void;
-}) => {
+};
+
+export const AssetSidebarNode = (props: AssetSidebarNodeProps) => {
+  const {node, level, toggleOpen, isOpen, selectThisNode} = props;
   const isGroupNode = 'groupNode' in node;
   const isLocationNode = 'locationName' in node;
   const isAssetNode = !isGroupNode && !isLocationNode;
 
-  const displayName = React.useMemo(() => {
-    if (isAssetNode) {
-      return getDisplayName(node);
-    } else if (isGroupNode) {
-      return node.groupNode.groupName;
-    } else {
-      return node.locationName;
-    }
-  }, [isAssetNode, isGroupNode, node]);
-
   const elementRef = React.useRef<HTMLDivElement | null>(null);
 
   const showArrow = !isAssetNode;
 
-  const ref = React.useRef<HTMLButtonElement | null>(null);
-  React.useLayoutEffect(() => {
-    // When we click on a node in the graph it also changes "isSelected" in the sidebar.
-    // We want to check if the focus is currently in the graph and if it is lets keep it there
-    // Otherwise it means the click happened in the sidebar in which case we should move focus to the element
-    // in the sidebar
-    if (ref.current && isLastSelected && !isElementInsideSVGViewport(document.activeElement)) {
-      ref.current.focus();
-    }
-  }, [isLastSelected]);
```

Comment:
This didn't get removed, just converted `GrayHoverBox` to a component and moved the hook down into it.